crest/Assets/Crest/Crest/Scripts/Helpers/UnderwaterPostProcess.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Rendering;
+
+namespace Crest
+{
+    [RequireComponent(typeof(Camera))]
+    public class UnderwaterPostProcess : MonoBehaviour
+    {
+        static int sp_OceanHeight = Shader.PropertyToID("_OceanHeight");
+        static int sp_HorizonOrientation = Shader.PropertyToID("_HorizonOrientation");
+        static int sp_MaskTex = Shader.PropertyToID("_MaskTex");
+        static int sp_MaskDepthTex = Shader.PropertyToID("_MaskDepthTex");
+        static int sp_InvViewProjection = Shader.PropertyToID("_InvViewProjection");
+
+        // TODO these should only be exposed as materials if the user is expected to want to tweak them? otherwise
+        // this script should just create materials by loading the shaders, as in animgerstnerwavebatched.cs
+        [Header("Materials")]
+        [Tooltip("Material used to render underwater fog in post-process."), SerializeField]
+        Material _underwaterPostProcessMaterial = null;
+        [Tooltip("Material used to re-render ocean to create a mask for underwater rendering."), SerializeField]
+        Material _oceanMaskMaterial = null;
+
+        [Header("Debug Options")]
+        public bool _viewOceanMask;
+        // end public debug options
+
+        private Camera _mainCamera;
+        private RenderTexture _textureMask;
+        private RenderTexture _depthBuffer;
+        private CommandBuffer _commandBuffer;
+        private PropertyWrapperMaterial _underwaterPostProcessMaterialWrapper;
+
+        // NOTE: We keep a list of ocean chunks to render for a given frame
+        // (which ocean chunks add themselves to) and reset it each frame by
+        // setting the currentChunkCount to 0. However, this could potentially
+        // be a leak if the OceanChunks are ever deleted. We don't expect this
+        // to happen, so this approach should be fine for now.
+        private List<Renderer> _oceanChunksToRender;
+        private int _oceanChunksToRenderCount;
+
+        // This matches const on shader side
+        private const float UNDERWATER_MASK_NO_MASK = 1.0f;
+        private const string FULL_SCREEN_EFFECT = "_FULL_SCREEN_EFFECT";
+        private const string DEBUG_VIEW_OCEAN_MASK = "_DEBUG_VIEW_OCEAN_MASK";
+
+
+        public void RegisterOceanChunkToRender(Renderer _oceanChunk)
+        {
+            if (_oceanChunksToRenderCount >= _oceanChunksToRender.Count)
```

Comment:
I don't think most of this is necessary. Simply Add and Remove from the list, and use the list Count. The list should do all of this efficiently. I can do this change, just validating my thinking here.

Reply:
I confirmed using dumb Add/Clear on list doesn't generate garbage, so replaced this